include/nbl/asset/ICPUComputePipeline.h

```diff
@@ -12,20 +12,21 @@ namespace nbl::asset
 {
 
 //! CPU Version of Compute Pipeline
-class ICPUComputePipeline : public ICPUPipeline<IPipeline<ICPUPipelineLayout>,1>
+class ICPUComputePipeline : public ICPUPipeline<IPipeline<ICPUPipelineLayout>>
 {
-        using base_t = ICPUPipeline<IPipeline<ICPUPipelineLayout>,1>;
+        using base_t = ICPUPipeline<IPipeline<ICPUPipelineLayout>>;
 
     public:
         struct SCreationParams final : IPipeline<ICPUPipelineLayout>::SCreationParams
         {
-            SShaderSpecInfo shader;
+            IPipelineBase::SShaderSpecInfo<true> shader;
         };
         static core::smart_refctd_ptr<ICPUComputePipeline> create(const SCreationParams& params)
         {
             if (!params.layout)
                 return nullptr;
             auto retval = new ICPUComputePipeline(core::smart_refctd_ptr<const ICPUPipelineLayout>(params.layout));
+            
             if (!retval->setSpecInfo(params.shader))
             {
                 retval->drop();
```

Comment:
get rid of the `SCreationParams` in the `ICPU..Pipeline`s, and lets have a public constructor which just takes a layout (default arg nullptr) and has null spec infos